### ISSUE TITLE
[Question] How can I enter Sort Order and Output Delimiter properties for TV's?

### ISSUE THREAD

**Issue by hugopeek** [theboxer/Git-Package-Management#45]
Or is this not (yet) possible?

**Comment by bartholomej** [comment]
This doesn't work yet.

**Comment by bartholomej** [comment]
@hugopeek: So this PR #48 added what you wanted. So did I... ;)

Config example:
```json
"sortOrder": "69",
"outputProperties": [{
    "delimiter": ","                    
}]
```

Need more testing... Let me know, please.

**Comment by hugopeek** [comment]
@bartholomej Cool! Sortorder is already working nicely!! Delimiter is not.. When I set it manually after package update, it's not being overwritten either, so nothing is happening there yet I think..

**Comment by Jako** [comment]
Try #51 or use the following:

```
"outputProperties": "[{\"delimiter\": \",\"}]
```

**Comment by Jako** [comment]
Maybe it is because of using index 0 in the following lines:
https://github.com/TheBoxer/Git-Package-Management/blob/e057a8a5cae185dfb03d495b5446717e34513f74/core/components/gitpackagemanagement/processors/mgr/gitpackage/create.class.php#L607
https://github.com/TheBoxer/Git-Package-Management/blob/e057a8a5cae185dfb03d495b5446717e34513f74/core/components/gitpackagemanagement/processors/mgr/gitpackage/buildpackage.class.php#L367
https://github.com/TheBoxer/Git-Package-Management/blob/e057a8a5cae185dfb03d495b5446717e34513f74/core/components/gitpackagemanagement/processors/mgr/gitpackage/update.class.php#L387

The index does not occur with input_properties.

**Comment by bartholomej** [comment]
OK, can anyone confirm this is really doesn't work? (@Jako)

I think the only problem is properties are not just being displayed properly (but delimiter is actually set).

**How to try?**
When you _update package_ via GPM and open `testtv` -> `Output Options` you will probably see `Default` (it's bug I know that).
Now change it manually to `Delimiter`. Can you see my testing value: `|pipe|`?

Can you confirm? (I'll fix that bug with displaying later)

Thanks

Use this test example:
```json
"tvs": [{
    "caption": "testTV",
    "name": "testtv",
    "type": "list",
    "description": "This is the best TV",
    "defaultValue": "second",
    "inputOptionValues": "First==first||Second==second",
    "templates": ["PackageTemplate"],
    "properties": [{
        "name": "testproperty",
        "value": "testvalue"
    }],
    "inputProperties": [{
        "allowBlank": false
    }],
    "outputProperties": [{
        "delimiter": "|pipe|"
    }]                
}],
```

**Comment by Jako** [comment]
inputProperties are not set with the current code. outputProperties are set. 

So your additions should be fine and `$tvObject->set('input_properties',$inputProperties);` should be changed to `$tvObject->set('input_properties',$inputProperties[0]);` in create & build processor.

A small difference between GPM and normal TV creation: boolean values are saved different. 
GPM saves `"allowBlank";b:0` and MODX saves `"allowBlank";s:5:"false"`. But both behave the same during editing the TV and a resource containing the TV.

**Comment by theboxer** [comment]
This should be fixed in cc8b1d887a162e403b57fdd1d7952f6cee6b0f49

Also I added `display` property: http://theboxer.github.io/Git-Package-Management/config/package/#template-variables
